src/Numerics/DGMethods/DGModel.jl

```diff
@@ -183,45 +153,23 @@ function (dg::DGModel)(tendency, state_prognostic, _, t, α, β)
     end
 
     if num_state_gradient_flux > 0 || nhyperviscstate > 0
-
-        comp_stream = volume_gradients!(device, (Nq, Nq))(
-            balance_law,
-            Val(dim),
-            Val(N),
-            dg.diffusion_direction,
-            state_prognostic.data,
-            state_gradient_flux.data,
-            Qhypervisc_grad.data,
-            state_auxiliary.data,
-            grid.vgeo,
-            t,
-            grid.D,
-            Val(hypervisc_indexmap),
-            topology.realelems,
-            ndrange = (Nq * nrealelem, Nq),
-            dependencies = (comp_stream,),
+        ########################
+        # Gradient Computation #
+        ########################
+
+        comp_stream = launch_volume_gradients!(
+            dg,
+            state_prognostic,
+            t;
+            dependencies = comp_stream,
```

Comment:
I love this idea and the way you implemented it below!

Reply:
I just took the idea (and commits) from #1592 and developed it further so credit goes to @charleskawczynski .